internal/datanode/channel_meta.go

```diff
@@ -179,12 +199,13 @@ func (c *ChannelMeta) addSegment(req addSegmentReq) error {
 		zap.Bool("importing", req.importing),
 	)
 	seg := &Segment{
-		collectionID: req.collID,
-		partitionID:  req.partitionID,
-		segmentID:    req.segID,
-		numRows:      req.numOfRows, // 0 if segType == NEW
-		startPos:     req.startPos,
-		endPos:       req.endPos,
+		collectionID:     req.collID,
+		partitionID:      req.partitionID,
+		segmentID:        req.segID,
+		numRows:          req.numOfRows, // 0 if segType == NEW
+		historyInsertBuf: make([]*BufferData, 0),
+		historyDeleteBuf: make([]*DelDataBuf, 0),
+		startPos:         req.startPos,
 	}
 	seg.sType.Store(req.segType)
```

Comment:
Please help to change this statement into `seg.setType(t)`

Reply:
done